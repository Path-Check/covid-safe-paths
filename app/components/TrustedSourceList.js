import findIndex from 'lodash/findIndex';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { StyleSheet, Switch, TouchableOpacity } from 'react-native';
import { MenuOption } from 'react-native-popup-menu';

import { Typography } from '../components/Typography';
import Colors from '../constants/colors';

export const TrustedSourceList = ({
  selectedAuthorities,
  authoritiesList,
  isAuthorityFilterActive,
  toggleFilterAuthoritesByGPSHistory,
  addAuthorityToState,
  filterAuthoritesByGPSHistory,
  onMenuClick,
}) => {
  const { t } = useTranslation();
  const isAuthoritySelected = item => {
    let name = Object.keys(item)[0];
    return findIndex(selectedAuthorities, ['key', name]) < 0;
  };
  const availableAuthorities = authoritiesList.filter(item =>
    isAuthoritySelected(item),
  );

  return (
    <>
      {__DEV__ && (
        <TouchableOpacity
          style={styles.authorityFilter}
          onPress={() => toggleFilterAuthoritesByGPSHistory()}>
          <Typography style={styles.authorityFilterText} use={'body2'}>
            {t('label.filter_authorities_by_gps_history')}
          </Typography>
          <Switch
            onValueChange={val => filterAuthoritesByGPSHistory({ val })}
            value={isAuthorityFilterActive}
          />
        </TouchableOpacity>
      )}
      {availableAuthorities &&
        availableAuthorities.map(item => {
          let name = Object.keys(item)[0];
          let key = authoritiesList.indexOf(item);

          return (
            <MenuOption
              key={key}
              onSelect={() => {
                addAuthorityToState(name);
              }}
              disabled={authoritiesList.length === 1}>
              <Typography style={styles.menuOptionText} use={'body2'}>
                {name}
              </Typography>
            </MenuOption>
          );
        })}
      <MenuOption onSelect={onMenuClick}>
        <Typography style={styles.menuOptionText} use={'body2'}>
          {t('label.authorities_add_url')}
        </Typography>
      </MenuOption>
    </>
  );
};

const styles = StyleSheet.create({
  authorityFilter: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 5,
    backgroundColor: Colors.LIGHT_GRAY,
    borderTopWidth: 3,
    borderTopColor: Colors.DIVIDER,
    justifyContent: 'space-between',
  },
  authorityFilterText: {
    padding: 10,
    color: Colors.VIOLET_TEXT,
  },
  menuOptionText: {
    padding: 10,
  },
});
