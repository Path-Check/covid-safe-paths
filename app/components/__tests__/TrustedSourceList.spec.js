import { render } from '@testing-library/react-native';
import React from 'react';

import { TrustedSourceList } from '../TrustedSourceList';

const trustedSourceList = [
  { first: [{ url: '' }, { bound: '' }] },
  { second: [{ url: '' }, { bound: '' }] },
];
const selected = [{ first: [{ url: '' }, { bound: '' }] }];

it('renders trusted source list', () => {
  const { asJSON } = render(
    <TrustedSourceList
      authoritiesList={trustedSourceList}
      selectedAuthorities={selected}
    />,
  );

  expect(asJSON()).toMatchSnapshot();
});
