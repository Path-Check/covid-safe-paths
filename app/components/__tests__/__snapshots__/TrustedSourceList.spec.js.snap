// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders trusted source list 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEEEEE",
        "borderTopColor": "#A5AFFB",
        "borderTopWidth": 3,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3C4ED8",
          "fontFamily": "IBMPlexSans",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 22,
          "padding": 10,
          "writingDirection": "ltr",
        }
      }
      use="body2"
    >
      Filter by your locations
    </Text>
    <RCTSwitch
      accessibilityRole="button"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={false}
    />
  </View>
  <MenuOption
    disabled={false}
  >
    <Text
      style={
        Object {
          "fontFamily": "IBMPlexSans",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 22,
          "padding": 10,
          "writingDirection": "ltr",
        }
      }
      use="body2"
    >
      first
    </Text>
  </MenuOption>
  <MenuOption
    disabled={false}
  >
    <Text
      style={
        Object {
          "fontFamily": "IBMPlexSans",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 22,
          "padding": 10,
          "writingDirection": "ltr",
        }
      }
      use="body2"
    >
      second
    </Text>
  </MenuOption>
  <MenuOption>
    <Text
      style={
        Object {
          "fontFamily": "IBMPlexSans",
          "fontSize": 16,
          "fontWeight": "normal",
          "lineHeight": 22,
          "padding": 10,
          "writingDirection": "ltr",
        }
      }
      use="body2"
    >
      Add authority via URL
    </Text>
  </MenuOption>
</View>
`;
